### PR TITLE
fix: InputNumber handleVisible token should work as expected

### LINKED ISSUE #51727: InputNumber token `handleVisible` doesn't work as expected
### Reproduction link

[Edit on CodeSandbox](https://codesandbox.io/p/sandbox/ji-ben-antd-5-22-2-forked-rlpq98?file=%2Fdemo.tsx%3A14%2C7)

### Steps to reproduce

View the demo

### What is expected?

image

### What is actually happening?

image

| Environment | Info |
| --- | --- |
| antd | 5.22.2 |
| React | 18.3.1 |
| System | MacOS |
| Browser | Chrome 120.0.6099.225 |

---
image

### PR DESCRIPTION
**Pull request by dengfuping**
[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE_CN.md?plain=1)

### 🤔 This is a ...

- [ ] 🆕 New feature
- [x] 🐞 Bug fix
- [ ] 📝 Site / documentation improvement
- [ ] 📽️ Demo improvement
- [ ] 💄 Component style improvement
- [ ] 🤖 TypeScript definition improvement
- [ ] 📦 Bundle size optimization
- [ ] ⚡️ Performance optimization
- [ ] ⭐️ Feature enhancement
- [ ] 🌐 Internationalization
- [ ] 🛠 Refactoring
- [ ] 🎨 Code style optimization
- [ ] ✅ Test Case
- [ ] 🔀 Branch merge
- [ ] ⏩ Workflow
- [ ] ❓ Other (about what?)

### 🔗 Related Issues

- Fix #51727

### 📝 Change Log

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | - 🐞 Fix InputNumber `handleVisible` token not work as expected.        |
| 🇨🇳 Chinese | - 🐞 修复 InputNumber `handleVisible` token 不生效的问题。          |

### CODE REVIEW

File: components/input-number/index.zh-CN.md [hunk 1]
```diff
@@ -31,6 +31,7 @@ demo:
 <code src="./demo/presuffix.tsx">前缀/后缀</code>
 <code src="./demo/status.tsx">自定义状态</code>
 <code src="./demo/focus.tsx" version="5.22.0">聚焦</code>
+<code src="./demo/handle-visible.tsx">操作按钮可见性</code>
```
Comment (afc163) [inline components/input-number/index.zh-CN.md:34]: 这个不适合做 demo，职责和 controls 的演示重复了。

File: components/input-number/style/index.ts [hunk 2]
```diff
@@ -250,7 +250,7 @@ const genInputNumberStyles: GenerateStyle<InputNumberToken> = (token: InputNumbe
           position: 'absolute',
           insetBlockStart: 0,
           insetInlineEnd: 0,
-          width: 0,
+          width: handleOpacity === 1 ? token.handleWidth : 0,
```
Comment (zombieJ) [inline components/input-number/style/index.ts:253]: 不能这么判断，在 cssVar 模式下，这里不是数字。Image Diff 失败也是因为这个

File: components/input-number/index.zh-CN.md (hunk 1 above)
Comment (dengfuping) [reply components/input-number/index.zh-CN.md:34]: 已去掉该 demo

File: components/input-number/style/index.ts (hunk 2 above)
Comment (dengfuping) [reply components/input-number/style/index.ts]: 已改成在注入 token 前进行判断

**Comment by afc163** [comment]
ConfigProvider 支持全局配置 controls 属性就行了吧。

**Comment by dengfuping** [comment]
> ConfigProvider 支持全局配置 controls 属性就行了吧。

试了下 `controls` 属性 https://codesandbox.io/p/sandbox/ji-ben-antd-5-22-2-forked-xdvc3m?file=%2Fdemo.tsx%3A17%2C5 并不是用于设置常驻展示增减按钮:

image

**Review by afc163 — APPROVED** [review]

**Pull request merged**